### PR TITLE
Release 4.8.10

### PR DESCRIPTION
**Pull request by m1kola**
### Which issue this PR addresses:

Work Item [№10405746](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/10405746/).

### What this PR does / why we need it:

Brings installer 4.8 into the RP.

### Test plan for issue:

* E2E run 
* Create a cluster locally

### Is there any documentation that needs to be updated for this PR?

No. User facing communications are being handled separately.

### CODE REVIEW

**Comment by m1kola** [comment]
I think https://github.com/openshift/installer/pull/4666 breakes our code: we were relying on defaults set here:

https://github.com/openshift/installer/blob/94125b21304a574ae5bc98039f8eb7f518293b83/pkg/asset/machines/master.go#L252-L253

But these defaults are not longer propogated to the installer config becuase this:

https://github.com/openshift/installer/blob/94125b21304a574ae5bc98039f8eb7f518293b83/pkg/asset/machines/master.go#L150

changed to this in 4.8:

https://github.com/openshift/installer/blob/c06e9b1562ccb76ac23163fb8db35f231ead6e97/pkg/asset/machines/master.go#L150

File: pkg/cluster/generateconfig.go [hunk 1]
```diff
@@ -142,6 +141,7 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 						EncryptionAtHost: m.doc.OpenShiftCluster.Properties.MasterProfile.EncryptionAtHost == api.EncryptionAtHostEnabled,
 						OSDisk: azuretypes.OSDisk{
 							DiskEncryptionSetID: m.doc.OpenShiftCluster.Properties.MasterProfile.DiskEncryptionSetID,
+							DiskSizeGB:          1024,
```
Comment (m1kola) [inline pkg/cluster/generateconfig.go:144]: https://github.com/openshift/installer/pull/4666 breakes our code: we were relying on defaults set here and the fact that installer was mutating the isntallconfig object:

https://github.com/openshift/installer/blob/94125b21304a574ae5bc98039f8eb7f518293b83/pkg/asset/machines/master.go#L252-L253

But these defaults are not longer propogated to the installer config becuase this:

https://github.com/openshift/installer/blob/94125b21304a574ae5bc98039f8eb7f518293b83/pkg/asset/machines/master.go#L150

changed to this in 4.8:

https://github.com/openshift/installer/blob/c06e9b1562ccb76ac23163fb8db35f231ead6e97/pkg/asset/machines/master.go#L150

Fixed it by adding default value to our `pkg/cluster/generateconfig.go`.

**Review by m1kola — COMMENTED** [review]
I recommend reviewing this commit-by-commit. I moved vendoring and generation of assets into separate commits.

I left few comments below which explain why we need some of ARO-RP changes.

File: pkg/metrics/statsd/golang/metrics.go [hunk 2]
```diff
@@ -34,11 +35,11 @@ func NewMetrics(log *logrus.Entry, m metrics.Interface) (Runnable, error) {
 		r:   prometheus.NewRegistry(),
 	}
 
-	if err := g.r.Register(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{})); err != nil {
+	if err := g.r.Register(collectors.NewProcessCollector(collectors.ProcessCollectorOpts{})); err != nil {
```
Comment (m1kola) [inline pkg/metrics/statsd/golang/metrics.go:38]: `github.com/prometheus/client_golang/prometheus` got updated and `prometheus.NewProcessCollector` while still exists - it is depricated. Our `golangci-lint` job complains about it.

**Comment by mjudeikis** [comment]
overall looks good! :)

**Comment by m1kola** [comment]
Previous e2e run failed, but it looked like a flake. Waiting for another e2e run, but I think this is ready for review now.

**Review by mjudeikis — DISMISSED** [review]

**Comment by m1kola** [comment]
I created a separate PR on top of this one to set min TLS version to 1.2 for storage accoutns to reflect what installer is doing: https://github.com/Azure/ARO-RP/pull/1730.

We need to merge https://github.com/Azure/ARO-RP/pull/1730 after this PR.

File: pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.5.0
+    controller-gen.kubebuilder.io/version: v0.6.2
```
Comment (bryanro92) [inline pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml:7]: sorry for my ignorance, where does this version come from?

**Review by bryanro92 — APPROVED** [review]
Looks good thanks for detailed explanation with links etc.. in comments. Much easier to understand with that information available.

File: pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml (hunk 3 above)
Comment (m1kola) [reply pkg/operator/deploy/staticresources/aro.openshift.io_clusters.yaml:7]: @bryanro92 Good question. I should've explained it too as it is not obvious.

We generate this yaml using `controller-gen` which is part of `sigs.k8s.io/controller-tools`. It got updated from `v0.5.0` to `v0.6.2` and as a result on `make generate` this yaml got updated too.

You can see version change of `sigs.k8s.io/controller-tools` in `go.mod`.

File: pkg/util/version/const.go [hunk 4]
```diff
@@ -27,8 +27,8 @@ var GitCommit = "unknown"
 
 // InstallStream describes stream we are defaulting to for all new clusters
 var InstallStream = &Stream{
-	Version:  NewVersion(4, 7, 21),
-	PullSpec: "quay.io/openshift-release-dev/ocp-release@sha256:60454c3270a61432447c752a9d5ee3e7d84cebcfd5d371eb4daa263623923dac",
```
Comment (jewzaam) [inline pkg/util/version/const.go:31]: 4.7 needs moved to `UpgradeStreams`

**Review by jewzaam — CHANGES_REQUESTED** [review]

File: pkg/util/version/const.go (hunk 4 above)
Comment (m1kola) [reply pkg/util/version/const.go:31]: Good catch! Updated.

**Review by jewzaam — APPROVED** [review]

**Pull request merged**